### PR TITLE
CI: do not pull PR Branch when testing a branch

### PR DESCRIPTION
**Pull request by chavafg**
When the CI runs on master (after merging a PR), the
resolve_dependencies.sh was trying to fetch an unexisting
(pr) branch from the remote. This change adds a validation
to only do this when testing a PR.

Fixes: #803.

Signed-off-by: Salvador Fuentes <salvador.fuentes@intel.com>

### CODE REVIEW

File: .ci/resolve-kata-dependencies.sh [hunk 1]
```diff
@@ -62,13 +62,14 @@ clone_repos() {
...
-		# to have latest changes of this repository, since the job starts. So we
-		# need to verify if we are already in the PR branch, before trying to
-		# fetch the same branch.
-		if [ ${repo} == ${tests_repo} ]
+		# When we have a change from the tests repo, the tests repository is cloned
+		# and checkout to the PR branch directly in the CI configuration (e.g. jenkins
+		# config file or zuul config), because we want to have latest changes
+		# of this repository since the job starts. So we need to verify if we
+		# are already in the PR branch, before trying to fetch the same branch.
+		if [ ${repo} == ${tests_repo} ] && [ "${repo}" == "${kata_repo}" ]
```
Comment (jodh-intel) [inline .ci/resolve-kata-dependencies.sh]: Nit: might be worth adding a comment on the first use of the `kata_repo` variable (line 18?) stating this is set by Jenkins.

**Comment by jodh-intel** [comment]
lgtm

File: .ci/resolve-kata-dependencies.sh (hunk 1 above)
Comment (grahamwhaley) [inline .ci/resolve-kata-dependencies.sh]: nit - whilst you are there, just below here, there looks like an indent syntax error - this line:

`current_branch=$(git rev-parse --abbrev-ref HEAD)`

needs indenting?

Comment (grahamwhaley) [reply .ci/resolve-kata-dependencies.sh]: ack - let's discuss/expand this a touch. The line confuses me a little. If I'm reading that right:
```
if [ ${repo} == ${tests_repo} ] && [ "${repo}" == "${kata_repo}" ]
```

then kata_repo must be the same as tests_repo for this to pass? I'm a bit confused about the difference between `repo` and `kata_repo`.

Comment (grahamwhaley) [reply .ci/resolve-kata-dependencies.sh]: :-) I pulled the source, and that line looks to be one tab short of an indent in my editor. Odd, as I don't see any space/tab mixing in the file either.

**Comment by grahamwhaley** [comment]
'nested' fails on CI - will wait until we are sure that is fixed before re-triggering.

**Comment by chavafg** [comment]
CI is green, @kata-containers/tests ptal

**Comment by GabyCT** [comment]
lgtm

**Pull request merged**